compiler/src/iree/compiler/Dialect/Stream/Transforms/ScheduleAllocation.cpp

```diff
@@ -390,15 +390,22 @@ struct AllocationScope {
       llvm::dbgs() << "\n";
     });
 
-    // TODO(#5410): make alias propagation map through an indexing map for
-    // slices/updates. Right now we assume all aliases are 1:1 full maps.
+    // Propagate alias subranges when present.
```

Comment:
And what about line 379? I never figured out if it needs and update as well?

Reply:
I _think_ it's ok (famous last words :) - all callers of mapResourceRange are explicitly setting the range and target resource with an offset/length if they have it - though looking into that made me notice something else that may need something so I'll try to write a test for it and see :)

Reply:
Yeah there's some issues with subviews but not in ways the compiler generates - they're really only intended to be supported on inputs and not outputs of the regions except for the special case you found with tied operands. We'll have to keep watch for different usage!